portfolio/src/main/webapp/script.js

```diff
@@ -85,16 +90,16 @@ function makeComment(heading, content) {
 
 function getDropdownVal() {
     const dropdown = document.getElementById("num-comments");
-    dropdown.onchange = changeDropdownVal;
-    if (sessionStorage["num-comments"]) {
-        dropdown.value = sessionStorage["num-comments"];
-    }   
+    const defaultComments = 10;
+    sessionStorage["num-comments"] = sessionStorage["num-comments"] || defaultComments;
+    dropdown.value = sessionStorage["num-comments"];
 }
 
 function changeDropdownVal() {
     const dropdown = document.getElementById("num-comments");
     sessionStorage["num-comments"] = dropdown.value;
-    this.form.submit();
+    sessionStorage["current-page"] = 1; // go back to the first page of comments
+    document.getElementById("dropdown").submit();
```

Comment:
I don't think this needs to submit to the server that the number of comments per page has changed. I think instead it should call getMessages() to update the visible comments.

Reply:
ohhh ok i see, I changed it to call getMessages instead!